public/app/plugins/datasource/grafana-azure-monitor-datasource/components/VariableEditor/VariableEditor.test.tsx

```diff
@@ -82,6 +83,53 @@ describe('VariableEditor:', () => {
     });
   });
 
+  describe('Azure Resource Graph queries:', () => {
+    const ARGqueryProps = (onChange?: (query: AzureMonitorQuery) => void) => ({
```

Comment:
weirdly enough, making that a constant failed the tests when ran together (but okay individually), making it a function fixes the issue.

Reply:
take a look to the createMockDatasource definition, the problem is that there, `datasource.azureMonitorDatasource.getSubscriptions` is defined as jest.fn().mockResolvedValue**Once**([]) but here you are setting the mock in `datasource.getSubscriptions`  so it's never executing the mock you are defining here.

IMO, the right fix here would be to change it in ArgQueryEditor to call `datasource.getSubscriptions`

Reply:
Also, an easier way to define this is:
```ts
    const ARGqueryProps = {
      ...defaultProps,
      query: {
        refId: 'A',
        queryType: AzureQueryType.AzureResourceGraph,
        azureResourceGraph: {
          query: 'Resources | distinct type',
          resultFormat: 'table',
        },
        subscriptions: ['sub'],
      },
    }
```